src/models/sssd_conf.py

```diff
@@ -1,34 +1,94 @@
+        # in such case exception should be raised
+        if Path(cls._backup_original).is_file():
+            logger.error("Backup of original sssd.conf already exists")
+            raise FileExistsError('backup file exists')
+        if Path(cls._backup_default).is_file():
+            logger.error("Default sssd.conf backup already exists")
+            raise FileExistsError('backup file exists')
+
+        try:
+            with cls._sssd_conf_path.open() as config:
+                parser.read_file(config)
+            copy2(cls._sssd_conf_path, cls._backup_original)        # BACKUP
+        except FileNotFoundError:
+            logger.info(f"sssd.conf not present in {cls._sssd_conf_path}")
+            logger.info("Creating sssd.conf based on the template")
+        with cls._template.open() as template:
+            logger.info("Updating sssd.conf with values from the template")
+            parser.read_file(template)
+        with cls._sssd_conf_path.open("w") as config:
+            parser.write(config)
+        copy2(cls._sssd_conf_path, cls._backup_default)
 
     @classmethod
     def set(cls, key: str, value: str, section: str = None):
         """
         Set value in SSSD config file. Parameter 'section' has to be specified.
         """
-        previous = cls._parser.get(section, key, fallback=None)
-        if previous is None:
+        parser = ConfigParser()
+        parser.optionxform = str
+
+        try:
+            with cls._sssd_conf_path.open() as config:
+                parser.read_file(config)
+        except FileNotFoundError:
+            logger.info(f"sssd.conf not present in {cls._sssd_conf_path}")
+            raise
+
+        if not parser.has_section(section):
+            logger.warning(f"Section {section} not present in sssd.conf")
+            logger.warning(f"Adding section {section} to sssd.conf")
+            parser.add_section(section)
+
+        if not parser.has_option(section, key):
             previous = "Not set"
-        cls._parser.set(section, key, value)
+        else:
+            previous = parser.get(section, key, fallback=None)
+
+        parser.set(section, key, value)
 
         logger.info(f"Value is change in {cls._sssd_conf_path}")
         logger.debug(f"Old value in section [{section}] {key}={previous}")
         logger.debug(f"New value in section [{section}] {key}={value}")
 
+        with cls._sssd_conf_path.open("w") as config:
+            parser.write(config)
+
     @classmethod
-    def save(cls):
-        with cls._sssd_conf_path.open("w") as f:
-            cls._parser.write(f)
+    def clean(cls):
+        """
+        Removes sssd.conf file in case it was created by this package or
+        restore original sssd.conf in case the file was modified.
+        """
+        if Path(cls._backup_original).is_file():
+            copy2(cls._backup_original, cls._sssd_conf_path)
+        else:
+            os.remove(cls._sssd_conf_path)
```

Comment:
pathlib.Path object has the method [`unlink`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.unlink) for removing the files. Please, use existing methods in pathlib